rhine-bayes/src/FRP/Rhine/Bayes.hs

```diff
@@ -11,30 +14,38 @@ import Control.Monad.Bayes.Weighted
 -- dunai
 import qualified Control.Monad.Trans.MSF.Reader as DunaiReader
 
--- dunai
-import Data.MonadicStreamFunction.Bayes (SoftEq)
+-- dunai-bayes
+import Data.MonadicStreamFunction.Bayes (SoftEq, similarity)
 import qualified Data.MonadicStreamFunction.Bayes as DunaiBayes
 
 -- rhine
 import FRP.Rhine
 
--- FIXME Does this haddock work?
--- | See 'Dunai.bayesFilter''
-bayesFilter' :: (MonadMeasure m, SoftEq sensor) =>
+bayesFilter' ::
+  (MonadMeasure m, SoftEq sensor) =>
   -- | model
   ClSF m cl input (sensor, state) ->
   -- | external sensor, data source
   ClSF m cl input sensor ->
   ClSF m cl input (sensor, state)
-bayesFilter' = DunaiBayes.bayesFilter'
+bayesFilter' model sensor = proc input -> do
+  output <- sensor -< input
+  estimatedState <- bayesFilter model -< (input, output)
+  returnA -< (output, estimatedState)
+
+{- | Condition on one output of a distribution.
 
--- FIXME Does this haddock work?
--- | See 'Dunai.bayesFilter'
-bayesFilter :: (MonadMeasure m, SoftEq sensor) =>
+   p(x,y | theta) ~> p(x | y, theta)
+-}
+bayesFilter ::
+  (MonadMeasure m, SoftEq sensor) =>
   ClSF m cl input (sensor, latent) ->
   -- | external sensor, data source
   ClSF m cl (input, sensor) latent
-bayesFilter = DunaiBayes.bayesFilter
+bayesFilter model = proc (input, measuredOutput) -> do
+  (estimatedOutput, estimatedState) <- model -< input
+  arrM score -< similarity estimatedOutput measuredOutput
```

Comment:
I still feel like, at least in the docs, it should be made clear that `bayesFilter` is a bit dangerous. What I mean is that if `similarity` is not chosen carefully, it will be different to the likelihood given by the observation model, so that your posterior, as specified by `bayesFilter` applied to the prior and data will not be denotationally correct. (I realize that in practice this may not be important in all situations, and disregard if I'm misunderstanding anything).

Reply:
Yes, I'll have to supply something like https://github.com/turion/dunai-bayes/issues/3#issuecomment-1382885938, I'm just not sure where I should put it.

Reply:
Intuitively, it seems like a change that needs to be first pioneered for monad-bayes, but I'm not sure either.

Reply:
Yes, agree. I'll release `rhine-bayes` without this, and we can go to the drawing board in `monad-bayes` without this particular application in mind.